programs/marinade-finance/src/instructions/management/redelegate.rs

```diff
@@ -0,0 +1,368 @@
+use crate::{
+    checks::check_stake_amount_and_validator,
+    error::MarinadeError,
+    state::{stake_system::StakeSystem, validator_system::ValidatorSystem},
+    State,
+};
+use std::convert::TryFrom;
+
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::{
+    program::invoke_signed,
+    stake::{self, state::StakeState},
+    system_program,
+};
+use anchor_spl::stake::{withdraw, Stake, StakeAccount, Withdraw};
+
+#[derive(Accounts)]
+pub struct ReDelegate<'info> {
+    #[account(mut)]
+    pub state: Box<Account<'info, State>>,
+    #[account(
+        address = state.validator_system.manager_authority
+            @ MarinadeError::InvalidValidatorManager
+    )]
```

Comment:
Redelegate must be permissionless because it is a part of stake-delta process. No manager authority

Reply:
done